src/test/java/uk/gov/companieshouse/api/strikeoffobjections/service/impl/ObjectionServiceTest.java

```diff
@@ -19,26 +24,34 @@
     private static final String COMPANY_NUMBER = "12345678";
     private static final String REQUEST_ID = "87654321";
     private static final String OBJECTION_ID = "87651234";
+    private static final LocalDateTime MOCKED_TIME_STAMP = LocalDateTime.of(2020, 2,2, 0, 0);
 
     @Mock
     ApiLogger apiLogger;
 
     @Mock
-    StrikeOffObjectionsRepository strikeOffObjectionsRepository;
+    ObjectionRepository objectionRepository;
+
+    @Mock
+    Supplier<LocalDateTime> localDateTimeSupplier;
 
     @InjectMocks
     ObjectionService objectionService;
 
     @Test
     void createObjectionTest() {
-        StrikeOffObjectionsEntity returnedEntity = new StrikeOffObjectionsEntity.Builder()
+        Objection returnedEntity = new Objection.Builder()
                 .withCompanyNumber(COMPANY_NUMBER)
                 .build();
         returnedEntity.setId(OBJECTION_ID);
-        when(strikeOffObjectionsRepository.save(any())).thenReturn(returnedEntity);
+        when(objectionRepository.save(any())).thenReturn(returnedEntity);
+        when(localDateTimeSupplier.get()).thenReturn(MOCKED_TIME_STAMP);
 
+        ArgumentCaptor<Objection> acObjection = ArgumentCaptor.forClass(Objection.class);
         String returnedId = objectionService.createObjection(REQUEST_ID, COMPANY_NUMBER);
 
+        verify(objectionRepository).save(acObjection.capture());
         assertEquals(OBJECTION_ID, returnedId);
+        assertEquals(MOCKED_TIME_STAMP, acObjection.getValue().getCreatedOn());
```

Comment:
also assert acObjection has had the COMPANY_NUMBER set on it (to ensure the number passed into createObjection is being set on the entity)